db/config.cc

```diff
@@ -880,7 +880,7 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , alternator_enforce_authorization(this, "alternator_enforce_authorization", value_status::Used, false, "Enforce checking the authorization header for every request in Alternator")
     , alternator_write_isolation(this, "alternator_write_isolation", value_status::Used, "", "Default write isolation policy for Alternator")
     , alternator_streams_time_window_s(this, "alternator_streams_time_window_s", value_status::Used, 10, "CDC query confidence window for alternator streams")
-    , alternator_timeout_in_ms(this, "alternator_timeout_in_ms", value_status::Used, 10000,
+    , alternator_timeout_in_ms(this, "alternator_timeout_in_ms", liveness::LiveUpdate, value_status::Used, 10000,
```

Comment:
Do I get this right, that this particular patch has nothing to do with `timeout_config` and it _just_ makes the `alternator_timeout_in_ms` live-updateable with all the necessary changed in alternator code to support the update?

Reply:
If I understand correctly, "timeout_config" was some sort of class that covered some random selection of timeout-related parameters in the configuration, and alternator_timeout_in_ms was never part of it.
To be honest, I don't understand why a timeout_config class was ever needed, it sounds to me that every piece of code needs to request one specific timeout, and there is never a need to get the entire structure. But maybe I'm missing something?

Reply:
We already have a "set of configuration parameters that Scylla supports", I don't see any reason why we need special subsets for it, like the "list of timeouts", the "list of ports", and so on. I think it just complicates everything. I think what we *should* have done is for each server to have a pointer to the configuration of the current shard - and the server could use this configuration to get anything from it. I think the current design is a mistake (but of course, no need to fix it in this PR, this is not a new issue).

In particular, if we don't want Scylla to become one huge spaghetti code, there is no reason to bundle CQL features, Redis features, and Alternator features together just because they sound superficially similar. Yes, CQL, Redis and Alternator all have some sort of timeout options - but these are *different* options, with different meanings, different defaults, etc., so bundling all of them together just makes the code harder to decipher.

Reply:
Yes, you're right. And we have per-service config to collect the needed parameters from db::config and pull them down the service. I guess the timeout_config was introduce to generalize this between transport, redis and thrift.

Reply:
> We already have a "set of configuration parameters that Scylla supports", I don't see any reason why we need special subsets for it, like the "list of timeouts", the "list of ports", and so on. I think it just complicates everything. I think what we _should_ have done is for each server to have a pointer to the configuration of the current shard - and the server could use this configuration to get anything from it. I think the current design is a mistake (but of course, no need to fix it in this PR, this is not a new issue).
> 
> In particular, if we don't want Scylla to become one huge spaghetti code, there is no reason to bundle CQL features, Redis features, and Alternator features together just because they sound superficially similar. Yes, CQL, Redis and Alternator all have some sort of timeout options - but these are _different_ options, with different meanings, different defaults, etc., so bundling all of them together just makes the code harder to decipher.

@nyh i think we can implement in a follow-up PR, which 

1. pushes the `updateable_timeout_config` down to their owner, namely, the config bundles in redis, thrift and CQL servers.
2. replace the bundles with a reference to `db::config`.
 
if we are all for this change.

Reply:
> I see. I would then suggest submitting this patch as separate PR

okay. but can we continue the discussion related to  `alternator_timeout_in_ms` , as @nyh already started review it and post some comments on it.